src/Geopilot.Api/Controllers/DeliveryController.cs

```diff
@@ -76,22 +77,42 @@ public async Task<IActionResult> Create(DeliveryRequest declaration)
             return NotFound($"Mandate with id <{declaration.MandateId}> not found.");
         }
 
-        if (!mandate.Organisations.SelectMany(u => u.Users).Any(u => u.Id == user.Id))
+        if (mandate.EvaluatePrecursorDelivery == FieldEvaluationType.NotEvaluated && declaration.PrecursorDeliveryId.HasValue)
         {
-            logger.LogTrace($"User <{user.AuthIdentifier}> is not authorized to create a delivery for mandate with id <{declaration.MandateId}>.");
-            return Unauthorized($"User is not authorized for mandate with id <{declaration.MandateId}>");
+            ModelState.AddModelError(nameof(declaration.PrecursorDeliveryId), "Precursor delivery is not allowed for this mandate.");
+        }
+        else if (mandate.EvaluatePrecursorDelivery == FieldEvaluationType.Required && !declaration.PrecursorDeliveryId.HasValue)
+        {
+            ModelState.AddModelError(nameof(declaration.PrecursorDeliveryId), "Precursor delivery is required for this mandate.");
         }
 
-        var precursorDelivery = declaration.PrecursorDeliveryId.HasValue ?
-            context.Deliveries
-            .Include(d => d.Mandate)
-            .Where(d => d.Mandate!.Id == mandate.Id)
-            .FirstOrDefault(d => d.Id == declaration.PrecursorDeliveryId) : null;
-
+        var precursorDelivery = mandate.Deliveries.FirstOrDefault(d => d.Id == declaration.PrecursorDeliveryId);
```

Comment:
Q: Kann es hier jemals mehr als zwei Einträge haben? Falls nicht: `SingleOrDefault()` verwenden, damit wir merken, wenn etwas in Schieflage ist.

Reply:
Nein die Id's dürften hier definitiv niemals doppelt sein. Ich habe gleich alle `First...` aufrufe korrigiert.